### PR TITLE
Enable strict mode

### PR DESCRIPTION
**Pull request by digeff**
I did the changes necessary to enable strict mode.

The most important changes are on chromeLauncher.ts which is being hidden because the diff is too large.
I refactored the code around, and put it into a few classes both to organize the code better, and to avoid having to check whether _chromePID, _chromeProc and userRequestedUrl are null...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,18 +25,22 @@
...
     "@types/mocha": "^2.2.48",
     "@types/chai": "^4.1.7",
     "@types/chai-string": "^1.4.1",
     "@types/mockery": "^1.4.29",
     "@types/node": "^8.10.39",
     "@types/puppeteer": "^1.12.3",
-    "@types/source-map": "^0.1.27",
+    "@types/request": "^2.48.1",
+    "@types/request-promise-native": "^1.0.15",
+    "@types/semver": "^6.0.0",
```
Comment (EricCornelson) [inline package.json]: What are these semver types for?

File: .vscode/launch.json [hunk 2]
```diff
@@ -80,7 +80,13 @@
             ],
             "skipFiles": [
                 "<node_internals>/**",
-                "methodCalledLogger.ts"
+                "methodCalledLogger.ts",
+                "runnable.js",
+                "runner.js",
+                "next_tick.js",
```
[the remark this answers is not in the document]
Comment (digeff) [reply .vscode/launch.json]: I was seeing them in the stack traces. "<node_internals>/**" was already  there. Is that supposed to happen?

File: package.json (hunk 1 above)
Comment (digeff) [reply package.json]: Request with promisify wasn't typing properly in strict mode, so I changed it to request-promise-native.
I don't think we need semver or request any more so I'll remove those.

File: .vscode/launch.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (roblourens) [reply .vscode/launch.json]: Not sure

**Pull request merged**